pkg/storage/chunk/prometheus_chunk.go

```diff
@@ -39,26 +74,11 @@ func (p *prometheusXorChunk) Add(m model.SamplePair) (EncodedChunk, error) {
 	return nil, nil
 }
 
-func (p *prometheusXorChunk) NewIterator(iterator Iterator) Iterator {
-	if p.chunk == nil {
-		return errorIterator("Prometheus chunk is not set")
-	}
-
-	if pit, ok := iterator.(*prometheusChunkIterator); ok {
-		pit.c = p.chunk
-		pit.it = p.chunk.Iterator(pit.it)
-		return pit
-	}
-
-	return &prometheusChunkIterator{c: p.chunk, it: p.chunk.Iterator(nil)}
-}
-
-func (p *prometheusXorChunk) Marshal(i io.Writer) error {
-	if p.chunk == nil {
-		return errors.New("chunk data not set")
-	}
-	_, err := i.Write(p.chunk.Bytes())
-	return err
+// AddHistogram adds another histogram to the chunk. While Add works, it is only implemented
+// to make tests work, and should not be used in production. In particular, it appends
+// all histograms to single chunk, and uses new Appender for each Add.
+func (p *prometheusXorChunk) AddHistogram(timestamp int64, h *histogram.Histogram) (EncodedChunk, error) {
+	panic("cannot add histogram to sample chunk")
```

Comment:
The comment doesn't correspond to the code.

Reply:
easy fix

Reply:
actually since this function can already return an error, it should just return that you cannot add a histogram value to a float chunk - which will always be the case as far as I can forsee